app/classifier/drawing-tools/rotate-rectangle.jsx

```diff
@@ -0,0 +1,292 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DrawingToolRoot from './root';
+import DragHandle from './drag-handle';
+import Draggable from '../../lib/draggable';
+import deleteIfOutOfBounds from './delete-if-out-of-bounds';
+import DeleteButton from './delete-button';
+
+const MINIMUM_SIZE = 5;
+const GUIDE_WIDTH = 1;
+const GUIDE_DASH = [4, 4];
+const DELETE_BUTTON_WIDTH = 8;
+const BUFFER = 16;
+
+export default class RotateRectangleTool extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getDeletePosition = this.getDeletePosition.bind(this);
+    this.handleMainDrag = this.handleMainDrag.bind(this);
+    this.handleTopLeftDrag = this.handleTopLeftDrag.bind(this);
+    this.handleTopRightDrag = this.handleTopRightDrag.bind(this);
+    this.handleBottomLeftDrag = this.handleBottomLeftDrag.bind(this);
+    this.handleBottomRightDrag = this.handleBottomRightDrag.bind(this);
+    this.handleRotate = this.handleRotate.bind(this);
+    this.normalizeMark = this.normalizeMark.bind(this);
```

Comment:
I'm not sure, but I don't think you need to bind `normalizeMark` here. Instance methods are bound to an instance of a class by default in javascript. Only event handlers need to be explicitly bound.

Reply:
I missed that it's also an event handler, for `onEnd`, which does change the `this` context.